src/object/object_representation.c

```diff
@@ -8197,6 +8084,105 @@ or_unpack_mvccid (char *ptr, MVCCID * mvccid)
 }
 
 /*
+ * or_mvcc_set_prev_version_lsa () - Set MVCC prev version LSA
+ *
+ * return	      : error code 
+ * buf (in/out)	      : or buffer
+ * mvcc_rec_header(in): MVCC record header
+ */
+STATIC_INLINE int
+or_mvcc_set_prev_version_lsa (OR_BUF * buf, MVCC_REC_HEADER * mvcc_rec_header)
+{
+  int error_code = NO_ERROR;
+
+  assert (buf != NULL);
+
+  ASSERT_ALIGN (buf->ptr, INT_ALIGNMENT);
+  if (!(mvcc_rec_header->mvcc_flag & OR_MVCC_FLAG_VALID_PREV_VERSION))
+    {
+      return NO_ERROR;
+    }
+
+  if ((buf->ptr + sizeof (LOG_LSA)) > buf->endptr)
+    {
+      return (or_overflow (buf));
+    }
+
+  memcpy (buf->ptr, &mvcc_rec_header->prev_version_lsa, sizeof (LOG_LSA));
+  buf->ptr += sizeof (LOG_LSA);
+
+  return NO_ERROR;
+}
+
+/*
+ * or_mvcc_get_prev_version_lsa () - Get MVCC prev version LSA from buffer
+ *
+ * return	        : error code 
+ * buf (in)	        : or buffer
+ * mvcc_flags(in)       : header mvcc flags
+ * prev_version_lsa(out): the LSA to previous version
+ * mvcc_rec_header(in)  : MVCC record header
+ */
+STATIC_INLINE int
+or_mvcc_get_prev_version_lsa (OR_BUF * buf, int mvcc_flags, LOG_LSA * prev_version_lsa)
+{
+  int error_code = NO_ERROR;
+  assert (buf != NULL);
+
+  ASSERT_ALIGN (buf->ptr, INT_ALIGNMENT);
+  if (!(mvcc_flags & OR_MVCC_FLAG_VALID_PREV_VERSION))
+    {
+      LSA_SET_NULL (prev_version_lsa);
+      return NO_ERROR;
+    }
+
+  if ((buf->ptr + sizeof (LOG_LSA)) > buf->endptr)
+    {
+      return (or_overflow (buf));
```

Comment:
one small flaw on object_representation.c:8141.
```
  if ((buf->ptr + sizeof (LOG_LSA)) > buf->endptr)
    {
      return (or_overflow (buf));
    }
````

`or_overflow` should be `or_underflow`.